hack/cluster-openshift.sh

```diff
@@ -178,12 +178,26 @@ OPENSHIFT_BIN_PATH="${OPENSHIFT_BIN_PATH:=${HOME}/bin}"
 # This is the IP address where OpenShift will bind its master.
 # This should be a valid IP address for the machine where OpenShift is installed.
 # NOTE: Do not use any IP address within the loopback range of 127.0.0.x.
-OPENSHIFT_IP_ADDRESS=${OPENSHIFT_IP_ADDRESS:-`echo $(ip -f inet addr | grep 'state UP' -A1 | tail -n1 | awk '{print $2}' | cut -f1 -d'/')`}
+if which ip > /dev/null ; then
+  OPENSHIFT_IP_ADDRESS=${OPENSHIFT_IP_ADDRESS:-`echo $(ip -f inet addr | grep 'state UP' -A1 | tail -n1 | awk '{print $2}' | cut -f1 -d'/')`}
+else
+  if [ ! "$OPENSHIFT_IP_ADDRESS" ] ; then
+    echo "ERROR: can't detect IP address please specify with --address argument"
+    exit 1
+  fi
+fi
 
 # The version is the tag from the openshift-istio/origin release builds.
 # The platform is either "linux" or "darwin".
 OS_ISTIO_OC_DOWNLOAD_VERSION="${OS_ISTIO_OC_DOWNLOAD_VERSION:-istio-3.9-0.8.0-alpha4}"
-OS_ISTIO_OC_DOWNLOAD_PLATFORM="${OS_ISTIO_OC_DOWNLOAD_PLATFORM:-linux}"
+DEFAULT_OS_VERSION=linux
+DETECTED_OS_VERSION=`uname | tr '[:upper:]' '[:lower:]'`
+if [ "${DETECTED_OS_VERSION}" = "linux" -o "${DETECTED_OS_VERSION}" = "darwin" ]
+then
+  DEFAULT_OS_VERSION=${DETECTED_OS_VERSION}
+  echo set default version to ${DEFAULT_OS_VERSION}
```

Comment:
Is this asking the user to "set" the default version to something? Seems like the user is being told to set something. I would make the message more clear - don't make it seem like a command to the user to do something; log it as an informative message. Something like:

`echo "The operating system has been detected as ${DEFAULT_OS_VERSION}"`

Reply:
I agree, it is confusing, probably better to use your text as set it as a debug message only